app/pages/home-for-user/news-pullout.jsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import apiClient from 'panoptes-client/lib/api-client';
+import ProjectCard from '../../partials/project-card';
+import Publications from '../about/publications';
+
+import style from './news-pullout.styl';
+void style;
+
+const NewsSection = React.createClass({
+
+  propTypes: {
+    newDatasets: React.PropTypes.array,
+  },
+
+  getDefaultProps() {
+    return {
+      newDatasets: [],
+    };
+  },
+
+  getInitialState() {
+    return {
+      avatars: {},
+      projects: [],
+      publications: [],
+    };
+  },
+
+  componentDidMount() {
+    this.fetchProjects();
+    this.recentPublications();
+  },
+
+  recentPublications() {
+    const articles = [];
+    Object.keys(Publications).forEach((category) => {
+      Publications[category].map((project) => {
+        return articles.push(...project.publications);
+      });
+    });
+    const newestPublications = articles.sort((a, b) => {
+      return new Date(b.date) - new Date(a.date);
+    }).splice(0, 3);
+    return this.setState({
+      publications: newestPublications,
+    });
+  },
+
+  fetchProjects() {
+    apiClient.type('projects').get({
+      sort: '-launch_date',
+      launch_approved: true,
+      page_size: 1,
+      include: ['avatar'],
+    })
+    .then((projects) => {
+      this.setState({
+        projects,
+      });
+
+      return Promise.all(projects.map((project) => {
+        return project.get('avatar')
+        .catch(() => {
+          return null;
+        })
+        .then((avatar) => {
+          this.state.avatars[project.id] = avatar;
```

Comment:
State should not be mutated directly like this: (see notes section about `setState`) https://facebook.github.io/react/docs/component-api.html

You can set the state of projects and avatars at the end of all of your promises.